js/gdm/loginDialog.js

```diff
@@ -688,12 +711,105 @@ const LoginDialog = new Lang.Class({
                                    this._onUserListActivated(item);
                                }));
 
+        this._customerSupportEmail = null;
+        this._customerSupportPhoneNumber = null;
    },
 
     _showPasswordHint: function() {
         this._passwordHintLabel.set_text(this._user.get_password_hint());
         this._passwordHintLabel.show();
         this._passwordHintButton.hide();
+        this._maybeShowPasswordResetButton();
+    },
+
+    _readCustomerSupportData: function() {
+        try {
+            let keyFile = new GLib.KeyFile();
+            keyFile.load_from_file(Config.PKGDATADIR + '/vendor-customer-support.ini', GLib.KeyFileFlags.NONE);
+            this._customerSupportEmail = keyFile.get_locale_string('Customer Support', 'Email', null);
+            this._customerSupportPhoneNumber = keyFile.get_locale_string('Customer Support', 'Phone', null);
+        } catch (e) {
+            logError(e, 'Failed to read customer support data');
+            return false;
+        }
+
+        return this._customerSupportEmail && this._customerSupportPhoneNumber;
+    },
+
+    _maybeShowPasswordResetButton: function() {
+        if (!this._customerSupportPhoneNumber || !this._customerSupportEmail) {
```

Comment:
You could rename `_readCustomerSupportData` to something like `_ensureCustomerSupportData` and fold this check in there.